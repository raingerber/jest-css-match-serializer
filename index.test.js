/* eslint-env jest */

const serializerFactory = require('./index')

describe('test with different option combinations', () => {
  const styles = `
    ${/* full match */''}
    #container {
      color: magenta;
    }
    ${/* partial match */''}
    #container-container > #container {
      color: purple;
      transform: scale(.5);
    }
    ${/* each part of the selector will be on a separate line
         in the snapshot; some of them are partial matches */''}
    ul li.aa,
    body > section li.bb,
    #container li:not(li) {
      opacity: .5;
    }
    ${/* should have a mediaText property in the snapshot */''}
    @media (max-width: 599px) {
      ul,
      li + li:nth-child(2n) {
        font-size: 12px;
        padding: 20px;
        margin: 30px;
      }
    }
  `
  const html = `
    <div id="container">
      <div class="xx">
        <ul class="yy">
          <li class="aa"></li>
          <li class="bb">
            <span>no css for me</span>
          </li>
          <li class="cc"></li>
        </ul>
      </div>
    </div>
  `
  const options = {
    recursive: false,
    includeHtml: false,
    includeCss: false,
    includeCssHash: false,
    includePartialMatches: false
  }
  const findMatches = serializerFactory(styles, options, expect)
  function executeFindMatchesTest (message, options) {
    it(message, async () => {
      expect(await findMatches(html, options)).toMatchSnapshot()
    })
  }
  executeFindMatchesTest(
    'all options === false',
    {}
  )
  executeFindMatchesTest(
    'recursive === true',
    {
      recursive: true
    }
  )
  // using "recursive" so the snapshots will have more data
  executeFindMatchesTest(
    'includePartialMatches === true',
    {
      recursive: true,
      includePartialMatches: true
    }
  )
  executeFindMatchesTest(
    'includeHtml === true',
    {
      recursive: true,
      includeHtml: true
    }
  )
  executeFindMatchesTest(
    'includeCss === true',
    {
      recursive: true,
      includeCss: true
    }
  )
  executeFindMatchesTest(
    'includeCssHash === true',
    {
      recursive: true,
      includeCssHash: true
    }
  )
  executeFindMatchesTest(
    'includeCssHash takes precedence over includeCss',
    {
      includeCss: true,
      includeCssHash: true
    }
  )
})

describe('html with multiple root elements', () => {
  it('create a snapshot with multiple roots', async () => {
    const html = `
      <div></div>
      <div>
        <div></div>
      </div>
      <div></div>
      <div>
        <span></span>
        <span></span>
      </div>
    `
    const styles = `
      div {
        color: red;
      }
    `
    const options = {
      recursive: true,
      includeHtml: true,
      includeCssHash: true,
      includePartialMatches: false
    }
    const findMatches = serializerFactory(styles, options, expect)
    expect(await findMatches(html, options)).toMatchSnapshot()
  })
})
