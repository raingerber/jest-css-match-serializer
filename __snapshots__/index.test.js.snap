// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`html with multiple root elements create a snapshot with multiple roots 1`] = `
[0]
<div>
div
fdb84ff02ab371841d8f7eb8a18b341e4010f20d

[1]
<div>
div
fdb84ff02ab371841d8f7eb8a18b341e4010f20d

  [0]
  <div>
  div
  fdb84ff02ab371841d8f7eb8a18b341e4010f20d

[2]
<div>
div
fdb84ff02ab371841d8f7eb8a18b341e4010f20d

[3]
<div>
div
fdb84ff02ab371841d8f7eb8a18b341e4010f20d

  [0]
  <span>
  <NULL>

  [1]
  <span>
  <NULL>
`;

exports[`test with different option combinations all options === false 1`] = `
[0]
#container
`;

exports[`test with different option combinations includeCss === true 1`] = `
[0]
#container
color: magenta

  [0]
  <NULL>

    [0]
    ul,
    li + li:nth-child(2n)
    font-size: 12px
    padding: 20px
    margin: 30px

      [0]
      ul li.aa,
      body > section li.bb,
      #container li:not(li)
      opacity: 0.5

      [1]
      ul,
      li + li:nth-child(2n)
      font-size: 12px
      padding: 20px
      margin: 30px

        [0]
        <NULL>

      [2]
      <NULL>
`;

exports[`test with different option combinations includeCssHash === true 1`] = `
[0]
#container
d762aa12d08b3083a1858ba276434c8b530f6152

  [0]
  <NULL>

    [0]
    ul,
    li + li:nth-child(2n)
    156afd3684d85b72cc907c7cc497f1ead048791b

      [0]
      ul li.aa,
      body > section li.bb,
      #container li:not(li)
      9149f5d9b4b160f75887e39c88e901e4b65ba5be

      [1]
      ul,
      li + li:nth-child(2n)
      156afd3684d85b72cc907c7cc497f1ead048791b

        [0]
        <NULL>

      [2]
      <NULL>
`;

exports[`test with different option combinations includeCssHash takes precedence over includeCss 1`] = `
[0]
#container
d762aa12d08b3083a1858ba276434c8b530f6152
`;

exports[`test with different option combinations includeHtml === true 1`] = `
[0]
<div id="container">
#container

  [0]
  <div class="xx">
  <NULL>

    [0]
    <ul class="yy">
    ul,
    li + li:nth-child(2n)

      [0]
      <li class="aa">
      ul li.aa,
      body > section li.bb,
      #container li:not(li)

      [1]
      <li class="bb">
      ul,
      li + li:nth-child(2n)

        [0]
        <span>
        <NULL>

      [2]
      <li class="cc">
      <NULL>
`;

exports[`test with different option combinations includePartialMatches === true 1`] = `
[0]
[4m#container[24m

#container-container > [4m#container[24m

  [0]
  <NULL>

    [0]
    [4mul[24m,
    li + li:nth-child(2n)

      [0]
      [4mul li.aa[24m,
      body > section li.bb,
      #container li:not(li)

      [1]
      ul li.aa,
      body > section [4mli.bb[24m,
      #container li:not(li)

      ul,
      [4mli + li:nth-child(2n)[24m

        [0]
        <NULL>

      [2]
      <NULL>
`;

exports[`test with different option combinations recursive === true 1`] = `
[0]
#container

  [0]
  <NULL>

    [0]
    ul,
    li + li:nth-child(2n)

      [0]
      ul li.aa,
      body > section li.bb,
      #container li:not(li)

      [1]
      ul,
      li + li:nth-child(2n)

        [0]
        <NULL>

      [2]
      <NULL>
`;
